gpib_server.py

```diff
@@ -176,10 +189,26 @@ def read(self, c, n_bytes=None):
         Otherwise, reads until the device stops sending.
         """
         instr = self.getDevice(c)
-        if n_bytes is None:
-            ans = instr.read_raw()
+        if instr.resource_name.startswith('USB'):
+            try:
+                if n_bytes is None:
+                    ans = instr.read()
+                else:
+                    ans = instr.read(n_bytes)
+            except:
+                self.asynchronous_wait(0.1)
+                if n_bytes is None:
+                    ans = instr.read()
+                else:
+                    ans = instr.read(n_bytes)
+            ans = str(ans)
+            if ",," in ans:
+                ans = ans.replace(",,", ",")  # rigol specific stupidity
```

Comment:
I added a clarifying comment in https://github.com/CampbellGroup/servers/commit/66b48359c2ac336a5ea292f17bec261f374338cc.

With this device connected we get the following with pyvisa 1.7.
```python
>>> import visa
>>> rm = visa.ResourceManager()
>>> rigol = rm.get_instrument('USB0::0x09C4::0x0400::DG1D150900538::INSTR')
>>> rigol.write('*IDN?')
>>> rigol.read()
u'RIGOL TECHNOLOGIES,DG1022 ,DG1D150900538,,00.03.00.09.00.02.11\n'
```

The double commas, before 00.03 are the use case we are taking care of.